actors/builtin/reward/reward_state.go

```diff
@@ -22,6 +22,9 @@ type State struct {
 	// and in advancing network time.
 	EffectiveNetworkTime abi.ChainEpoch
 
+	// EffectiveBaselinePower is the baseline power at the EffectiveNetworkTime epoch
+	EffectiveBaselinePower abi.StoragePower
```

Comment:
I don't understand this well enough to approve it. I don't think this matches how I'm expecting it to behave, but I also can't quite tell, and also my expectations might be wrong. We probably need a conversation to resolve this to both our satisfaction, though it's likely that a large part of that conversation could be replaced with excessive documentation of intent and algorithm's here.

- What does `effective` mean in this name? For the network time, it's effective rather than real because it's renaming power progress as time, it's not real time. But my understanding of the baseline function is that it should be a function of real network time/age in epochs.
- I thought that the baseline (target) power was an _input_ to the calculation of effective network time. The comment here makes that very confusing. If my interpretation is correct, it would be slightly better to put this in state before the EffectiveNetworkTime, and describe the calculation of one from the other.
- I understand the need for this state as purely an optimization around computing the exponential function `BaselineExponent^st.Epoch`. Baseline is fundamentally a function of real network time.

Reply:
I might be just confused. I see things kinda like this in documents elsewhere. The documentation present is just fundamentally not sufficient for me to have a clear conceptual understanding of how this is supposed to work, and so I can't verify it. I'm sure the phrase "inverse function" should appear somewhere. Maybe you could copy enough documentation from elsewhere such that this makes sense?

Reply:
This is BaselinePower at EffectiveNetworkTime. Previously we had straight-up formula: `BaselinePowerAt(st.EffectiveNetworkTime)` but now we have to compute it iteratively.

Reply:
Maybe this will make it clearer, we need to keep track of two BaselinePowers:
 - `BaselinePower(rt.Epoch)`
 - `BaselinePower(st.EffectiveNetworkTime)`